app/mailers/references_header_builder.rb

```diff
@@ -0,0 +1,94 @@
+# Builds RFC 5322 compliant References headers for email threading
+#
+# This module provides functionality to construct proper References headers
+# that maintain email conversation threading according to RFC 5322 standards.
+module ReferencesHeaderBuilder
+  # Builds a complete References header for an email reply
+  #
+  # According to RFC 5322, the References header should contain:
+  # - References from the message being replied to (if available)
+  # - The In-Reply-To message ID as the final element
+  # - Proper line folding if the header exceeds 998 characters
```

Comment:
Can we consider adding a test for very long reference chains

Reply:
The header has no size limit as such, only line limits. That won't be an issue since we are folding each ID into a separate line by default, so we will never touch the 998 character limit